src/webwatch/ww.go

```diff
@@ -17,81 +18,142 @@ import (
 	"time"
 )
 
-// report adds a message (printf style) to message to be emailed
-func report(msg *string, format string, values ...interface{}) {
-	add := fmt.Sprintf(format+"\n", values...)
-	log.Printf(add)
-	*msg += add
-}
+var (
+	url, warn, from, to, smtpServer *string
+	recipients                      []string
+	client                          *http.Client
+)
 
-// sendReport sends any report of whois differences via email
-func sendReport(server, from, warn, url, msg string, to []string) {
-	if msg == "" {
-		return
+func main() {
+	err := parseAndValidateConfiguration()
+	if err != nil {
+		log.Fatalln(err)
 	}
 
-	t := strings.Join(to, ", ")
-	header := fmt.Sprintf(`From: %s
-To: %s
-Date: %s
-Subject: WARNING! String %s found in URL %s
+	client = &http.Client{
+		Timeout: time.Duration(5 * time.Second),
+	}
 
-`, from, t, time.Now().Format(time.RFC822Z), warn, url)
+	body, err := fetchAndReturnPage()
 
-	msg = header + msg
-	err := smtp.SendMail(server, nil, from, to, []byte(msg))
-	if err != nil {
-		log.Printf("Error sending message from %s to %s via %s: %s",
-			from, t, server, err)
+	if strings.Contains(body, *warn) {
+		err = sendReportWithMessage("%q FOUND in %s", *warn, *url)
+		if err != nil {
+			log.Fatalln(err)
+		}
+	} else {
+		log.Printf("%q NOT found in %s\n", *warn, *url)
 	}
+
 }
 
-func main() {
-	url := flag.String("url", "", "URL to check")
-	warn := flag.String("warn", "",
+func parseAndValidateConfiguration() error {
+	url = flag.String("url", "",
+		"URL to check")
+	warn = flag.String("warn", "",
 		"Send email if this string is found in the web page")
-	from := flag.String("from", "", "Email addresses to send from")
-	to := flag.String("to", "",
+	from = flag.String("from", "",
+		"Email addresses to send from")
+	to = flag.String("to", "",
 		"Comma-separated list of email addresses to send to")
-	smtpServer := flag.String("smtp", "gmail-smtp-in.l.google.com:25",
+	smtpServer = flag.String("smtp", "gmail-smtp-in.l.google.com:25",
 		"Address of SMTP server to use (host:port)")
+
 	flag.Parse()
 
 	if *warn == "" {
-		log.Fatalf("The -warn parameter is required")
+		return errors.New("The -warn parameter is required")
 	}
 	if *to == "" {
-		log.Fatalf("The -to parameter is required")
+		return errors.New("The -to parameter is required")
 	}
 	if *from == "" {
-		log.Fatalf("The -from parameter is required")
+		return errors.New("The -from parameter is required")
 	}
 	if *url == "" {
-		log.Fatalf("The -url parameter is required")
+		return errors.New("The -url parameter is required")
 	}
 
+	err := checkIfSMTPURLIsValid()
+	if err != nil {
+		return err
+	}
+
+	parseRecipients()
```

Comment:
I find it clearer and more maintainable when a function does one thing. It might be a bit redundand here, given the size of the code. In the future, it might save some time (especially the ```checkIfSMTPURLIsValid()```). The other two are a bit redundant.